### ISSUE TITLE
Is y_pred defined in loss compution class label or score?

### ISSUE THREAD

**Issue by auspicious3000** [fchollet/keras#3351]
For a classification problem, what exactly is the y_pred defined in binary_crossentropy?
Is it class label like 01001100…… or score given by the top layer like 0.1 0.9 0.1 0.1 0.8 0.7 0.2 0.2……
In order to make the loss differentiable, it has to be the scores,  but when I used it in my custom loss function:

`def weighted_binary_crossentropy(y_true, y_pred, weights):`
    `mask = K.switch(K.equal(y_pred[:,:,0]-y_true[:,:,0], 1), K.ones_like(y_pred[:,:,0])*weights, 1)`
    `return K.mean(K.binary_crossentropy(y_pred, y_true), axis=-1)*mask`

This `K.equal(y_pred[:,:,0]-y_true[:,:,0], 1)` works as if y_pred is class label, which makes me confused. 

BTW, Are there ways to view this y_pred during training using callback or something?

Please help! Thanks in advance!
